internal/app/server/repository/postgres.go

```diff
@@ -0,0 +1,151 @@
+package repository
+
+import (
+	"context"
+	"errors"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+
+	"github.com/Te8va/APIbook/internal/app/server/domain"
+)
+
+type postgres struct {
+	*pgxpool.Pool
+}
+
+func NewPostgresBookRepository(pg *pgxpool.Pool) *postgres {
+	return &postgres{Pool: pg}
+}
+
+func (p *postgres) GetBookByID(id string) (domain.Book, error) {
+	var book domain.Book
+
+	err := p.QueryRow(context.Background(), "SELECT id, title, author, year, status FROM books WHERE id = $1", id).Scan(&book.ID, &book.Title, &book.Author, &book.Year, &book.Status)
+	if err != nil {
+		if err == pgx.ErrNoRows {
+			return domain.Book{}, domain.ErrBookNotFound
+		}
+		return domain.Book{}, domain.ErrReadingDatabase
+	}
+
+	if book.Status == "deleted" {
+		return domain.Book{}, domain.ErrDeletedBook
+	}
+
+	return book, nil
+}
+
+func (p *postgres) AddBook(ctx context.Context, newBook domain.Book) (string, error) {
+	err := p.UseTransaction(ctx, func(tx pgx.Tx) error {
+
+		newUUID := uuid.New()
+		newBook.ID = newUUID.String()
+
+		_, err := tx.Exec(ctx, "INSERT INTO books (id, title, author, year, status) VALUES ($1, $2, $3, $4, $5)", newBook.ID, newBook.Title, newBook.Author, newBook.Year, "")
+		if err != nil {
+			return domain.ErrDatabaseOperation
+		}
+		return nil
+	})
+	if err != nil {
+		return "", err
```

Comment:
you can wrap it to add additional info (for example, a layer where it was found) with fmt.Errorf